bindings/src/main/java/io/pravega/storage/hdfs/HDFSStorage.java

```diff
@@ -391,43 +492,25 @@ public void write(SegmentHandle handle, long offset, InputStream data, int lengt
     public SegmentHandle openWrite(String streamSegmentName) throws StreamSegmentException {
         ensureInitializedAndNotClosed();
         long traceId = LoggerHelpers.traceEnter(log, "openWrite", streamSegmentName);
-        long fencedCount = 0;
-        do {
```

Comment:
You need to keep this loop in here. The reason is that multiple instances of the container may try to open-write a file.

For example:
- Consider a file with epoch A.
- Container instance B with epoch B (B>A) attempts to open-write the file
- Container instance C with epoch C (C>B) attempts to open-write the file *at the same time as B*
- B succeeds in renaming the file before C (C has already executed line 498 (`findStatusForSegment`) so it knows the file is with epoch A. 
    - C will incorrectly fail and B acquires the write lock for this file. 

In this scenario, we need to run in a loop, as long as the file epoch is smaller than our epoch. If in the meantime the file epoch is larger than our's then bail out with `StorageWriterNotPrimaryException`

Reply:
Earlier change was trying to do checks lazily. The newer change does not make changes to current design. fixed.